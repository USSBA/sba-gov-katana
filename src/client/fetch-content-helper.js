import axios from 'axios'
import path from 'path'
import queryString from 'querystring'

async function fetchRestContent(id, langOverride) {
  let data = null

  try {
    const response = await axios.get(
      '/api/content' + (id ? '/' + id : '') + '.json',
      langOverride && { headers: { 'accept-language': langOverride } }
    )

    if (langOverride === 'es') {
      data = response.data.spanishTranslation
    } else {
      data = response.data
    }
  } catch (error) {
    console.error('fetchRestContent', error)
  }

  return data
}

async function fetchApiDistrictOfficeName(zipcode) {
  let data = null
  try {
    const response = await axios.get(
      `https://h4z2sm4z28.execute-api.us-east-1.amazonaws.com/default/dev-ryan-district-office-zipcode-api?zipcode=${zipcode}`
    )
    data = response.data
  } catch (error) {
    console.error('fetchDistrictOfficeName', error)
  }
  console.log('the data is', data)
  return data
}

async function fetchApiSbaGovContent(fileDirectory) {
  let data = null

  try {
    const response = await axios.get(`https://api.sba.gov/${fileDirectory}`)

    data = JSON.parse(response.data)
  } catch (error) {
    console.error('fetchApiSbaGovContent', error)
  }

  return data
}

async function fetchEventContent(id, langOverride) {
  let data = null

  try {
    const response = await axios.get(
      '/api/content/search/event' + (id ? '/' + id : '') + '.json',
      langOverride && { headers: { 'accept-language': langOverride } }
    )
    data = response.data
  } catch (error) {
    console.error('fetchEventContent', error)
  }

  return data
}

async function fetchSiteContent(type, query, langOverride) {
  let data = null

  try {
    const response = await axios.get(
      '/api/content/search/' + type + '.json' + (query ? '?' + queryString.stringify(query) : ''),
      langOverride && { headers: { 'accept-language': langOverride } }
    )
    data = response.data
  } catch (error) {
    console.error('fetchSiteContent', error)
  }
  return data
}

async function postMiscAction(endpoint, body) {
  const url = path.join('/actions/misc', `${endpoint}`)

  let data = null

  try {
    const response = await axios.post(url, body)
    data = response.data
  } catch (error) {
    console.error('postMiscAction', error)
    throw error
  }

  return data
}

async function runMiscAction(type, query) {
  const url = '/actions/misc/' + type + '.json' + (query ? '?' + queryString.stringify(query) : '')
  let data = null
  try {
    const response = await axios.get(url)
    data = response.data
  } catch (error) {
    console.error('fetchSiteContent', error)
  }
  return data
}

export {
  fetchApiDistrictOfficeName,
  fetchApiSbaGovContent,
  fetchRestContent,
  fetchEventContent,
  fetchSiteContent,
  postMiscAction,
  runMiscAction
}
