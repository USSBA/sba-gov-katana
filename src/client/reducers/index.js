import {combineReducers} from 'redux';
import {routerReducer} from 'react-router-redux';

// our customer reducers
import loanReducer from '../reducers/loan-form.js';
import additionalInfoReducer from '../reducers/additional-info.js';
import contactInfoReducer from '../reducers/contact-info.js';
import businessInfoReducer from '../reducers/business-info.js';
import reviewSubmitInfoReducer from './review-submit-info.js';
import industryInfoReducer from './industry-info.js';

// combine the custom reducers with the routing reducer
const rootReducer = combineReducers({
  loanReducer: loanReducer,
  additionalInfoReducer: additionalInfoReducer,
  contactInfoReducer: contactInfoReducer,
  businessInfoReducer: businessInfoReducer,
  reviewSubmitInfoReducer: reviewSubmitInfoReducer,
  industryInfoReducer: industryInfoReducer,
  routing: routerReducer
});

export default rootReducer;
