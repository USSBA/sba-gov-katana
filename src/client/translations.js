/* eslint max-lines: ["error", 1000] */
// eslint-disable-next-line id-match
const TRANSLATIONS = {
  // The words to translate are the keys:
  aboutSba: {
    en: {
      text: 'About SBA',
      url: '/about-sba'
    },
    es: {
      text: 'Acerca de la SBA',
      url: '/about-sba'
    }
  },
  aboutSbaGov: {
    en: {
      text: 'About SBA.gov'
    },
    es: {
      text: 'Acerca de SBA.gov'
    }
  },
  accessibility: {
    en: {
      text: 'Accessibility',
      url: '/about-sba/sba-performance/open-government/about-sbagov-website/accessibility'
    },
    es: {
      text: 'Accesibilidad',
      url: '/about-sba/sba-performance/open-government/about-sbagov-website/accessibility'
    }
  },
  adminTool: {
    en: {
      text: 'Admin Tool',
      url: '/admintool'
    },
    es: {
      text: 'Administrar',
      url: '/admintool'
    }
  },
  advocacy: {
    en: {
      text: 'Advocacy',
      url: '/advocacy'
    },
    es: {
      text: 'Defensa',
      url: '/advocacy'
    }
  },
  allTopics: {
    en: {
      text: 'all topics'
    },
    es: {
      text: 'todos los temas'
    }
  },
  banking: {
    en: {
      text: 'Banking'
    },
    es: {
      text: 'Cuenta bancaria'
    }
  },
  backTo: {
    en: {
      text: 'Back to'
    },
    es: {
      text: 'Regresar a'
    }
  },
  backToAllTopics: {
    en: {
      text: 'Back to all topics'
    },
    es: {
      text: 'Regresar a todos los temas'
    }
  },
  blog: {
    en: {
      text: 'Blog',
      url: '/blogs'
    },
    es: {
      text: 'Blog',
      url: '/blogs'
    }
  },
  businessGuide: {
    en: {
      text: 'Business Guide'
    },
    es: {
      text: 'Guia de negocios'
    }
  },
  close: {
    en: 'Close',
    es: 'Cerrar'
  },
  contactSba: {
    en: {
      text: 'Contact SBA',
      url: '/about-sba/what-we-do/contact-sba'
    },
    es: {
      text: 'Contacte con la SBA',
      url: '/about-sba/what-we-do/contact-sba'
    }
  },
  contactUs: {
    en: {
      text: 'Contact Us',
      url: '/about-sba/what-we-do/contact-sba'
    },
    es: {
      text: 'Contáctenos',
      url: '/acerca-de-la-sba/organizacion/comuniquese-la-sba'
    }
  },
  content: {
    en: 'Content',
    es: 'Contenidos'
  },
  customerService: {
    en: {
      text: 'Customer Service'
    },
    es: {
      text: 'Atención al cliente'
    }
  },
  dataStore: {
    en: {
      text: 'Data Store',
      url: '/about-sba/sba-performance/open-government/digital-sba/open-data'
    },
    es: {
      text: 'Almacenamiento de datos',
      url: '/about-sba/sba-performance/open-government/digital-sba/open-data'
    }
  },
  disasterAssistance: {
    en: {
      text: 'Disaster Assistance'
    },
    es: {
      text: 'Asistencia por desastre'
    }
  },
  disclaimers: {
    en: {
      text: 'Disclaimers',
      url: '/about-sba/sba-performance/open-government/about-sbagov-website/disclaimer'
    },
    es: {
      text: 'Avisos legales',
      url: '/about-sba/sba-performance/open-government/about-sbagov-website/disclaimer'
    }
  },
  eliminatingFraudWasteAndAbuse: {
    en: {
      text: 'Eliminating Fraud, Waste, and Abuse',
      url: '/about-sba/sba-performance/policy-regulations/eliminating-fraud-waste-abuse'
    },
    es: {
      text: 'Eliminación de fraude, despilfarro y abuso',
      url: '/about-sba/sba-performance/policy-regulations/eliminating-fraud-waste-abuse'
    }
  },
  federalContracting: {
    en: {
      text: 'Federal Contracting'
    },
    es: {
      text: 'Contratación federal'
    }
  },
  findCounselors: {
    en: {
      text: 'Find counselors'
    },
    es: {
      text: 'Encontrar asesores'
    }
  },
  findEvents: {
    en: {
      text: 'Find Events',
      url: '/events/find/'
    },
    es: {
      text: 'Encuentre eventos',
      url: '/events/find/'
    }
  },
  foia: {
    en: {
      text: 'FOIA',
      url: '/about-sba/sba-performance/open-government/foia'
    },
    es: {
      text: 'FOIA',
      url: '/about-sba/sba-performance/open-government/foia'
    }
  },
  forPartners: {
    en: {
      text: 'For Partners',
      url: '/partners'
    },
    es: {
      text: 'Para los socios',
      url: '/para-los-socios'
    }
  },
  fund: {
    en: {
      text: 'Fund'
    },
    es: {
      text: 'Financie'
    }
  },
  fundingPrograms: {
    en: {
      text: 'Funding Programs'
    },
    es: {
      text: 'Programas de financiamiento'
    }
  },
  freedomOfInformationAct: {
    en: {
      text: 'Freedom of Information Act',
      url: '/about-sba/sba-performance/open-government/foia'
    },
    es: {
      text: 'Ley de Libertad de Información',
      url: '/about-sba/sba-performance/open-government/foia'
    }
  },
  govDelivery: {
    en: {
      footerTitle: 'Sign up for SBA email updates',
      buttonText: 'Subscribe'
    },
    es: {
      footerTitle: 'Reciba las noticias de SBA por email',
      buttonText: 'Suscríbase'
    }
  },
  hearingsAndAppeals: {
    en: {
      text: 'Hearings and Appeals',
      url: '/oha'
    },
    es: {
      text: 'Audiencias y apelaciones',
      url: '/oha'
    }
  },
  initiatives: {
    en: {
      text: 'Initiatives',
      url: '/about-sba/sba-initiatives'
    },
    es: {
      text: 'Iniciativas',
      url: '/about-sba/sba-initiatives'
    }
  },
  inspectorGeneral: {
    en: {
      text: 'Inspector General',
      url: '/office-of-inspector-general'
    },
    es: {
      text: 'Inspector general',
      url: '/office-of-inspector-general'
    }
  },
  learningCenter: {
    en: {
      text: 'Learning Center',
      url: '/learning-center'
    },
    es: {
      text: 'Centro de capacitación',
      url: '/learning-center'
    }
  },
  learnMore: {
    en: {
      text: 'Learn more'
    },
    es: {
      text: 'Más información'
    }
  },
  license: {
    en: {
      text: 'License'
    },
    es: {
      text: 'Licencias'
    }
  },
  linkingPolicy: {
    en: {
      text: 'Linking Policy',
      url: '/about-sba/sba-performance/open-government/about-sbagov-website/linking-policy'
    },
    es: {
      text: 'Política de enlaces',
      url: '/about-sba/sba-performance/open-government/about-sbagov-website/linking-policy'
    }
  },
  localAssistance: {
    en: {
      text: 'Local Assistance',
      url: '/tools/local-assistance'
    },
    es: {
      text: 'Asistencia local',
      url: '/tools/local-assistance'
    }
  },
  location: {
    en: {
      text: 'Location'
    },
    es: {
      text: 'Ubicación'
    }
  },
  lookUpYourState: {
    en: {
      text: 'Look up your state'
    },
    es: {
      text: 'Encuentra tu estado'
    }
  },
  mediaAndPressRelations: {
    en: {
      text: 'Media and Press Relations',
      url: '/about-sba/sba-newsroom'
    },
    es: {
      text: 'Relaciones con los medios y la prensa',
      url: '/about-sba/sba-newsroom'
    }
  },
  myAccount: {
    en: {
      text: 'My Account',
      url(id) {
        return `/user/${id}/edit`
      }
    },
    es: {
      text: 'Mi cuenta',
      url(id) {
        return `/user/${id}/edit`
      }
    }
  },
  name: {
    en: {
      text: 'Name'
    },
    es: {
      text: 'Nombre'
    }
  },
  newsroom: {
    en: {
      text: 'Newsroom',
      url: '/about-sba/sba-newsroom'
    },
    es: {
      text: 'Sala de prensa',
      url: '/about-sba/sba-newsroom'
    }
  },
  next: {
    en: {
      text: 'Next'
    },
    es: {
      text: 'Siguiente'
    }
  },
  noFearAct: {
    en: {
      text: 'No Fear Act',
      url: '/about-sba/sba-performance/open-government/about-sbagov-website/no-fear-act'
    },
    es: {
      text: 'Ley de Notificación y Antidiscriminación y Represalias para Empleados Federales (ley No-FEAR)',
      url: '/about-sba/sba-performance/open-government/about-sbagov-website/no-fear-act'
    }
  },
  ombudsman: {
    en: {
      text: 'Ombudsman',
      url: '/ombudsman'
    },
    es: {
      text: 'Defensor general',
      url: '/ombudsman'
    }
  },
  onlineTraining: {
    en: {
      text: 'Online Training',
      url: '/tools/sba-learning-center/search/training'
    },
    es: {
      text: 'Capacitación en línea',
      url: '/tools/sba-learning-center/search/training'
    }
  },
  openGovernment: {
    en: {
      text: 'Open Government',
      url: '/about-sba/sba-performance/open-government'
    },
    es: {
      text: 'Gobierno abierto',
      url: '/about-sba/sba-performance/open-government'
    }
  },
  oversight: {
    en: {
      text: 'Oversight'
    },
    es: {
      text: 'Supervisión'
    }
  },
  oversightAndAdvocacy: {
    en: {
      text: 'Oversight & Advocacy',
      url: '/about-sba/oversight-advocacy'
    },
    es: {
      text: 'Supervisión y defensa',
      url: '/about-sba/oversight-advocacy'
    }
  },
  performanceBudgetAndFinancing: {
    en: {
      text: 'Performance, Budget, and Financing',
      url: '/about-sba/sba-performance/performance-budget-finances'
    },
    es: {
      text: 'Desempeño, presupuesto y financiamiento',
      url: '/about-sba/sba-performance/performance-budget-finances'
    }
  },
  plainLanguage: {
    en: {
      text: 'Plain Language',
      url: '/about-sba/sba-performance/open-government/other-plans-reports/plain-language-page'
    },
    es: {
      text: 'Lenguaje claro',
      url: '/about-sba/sba-performance/open-government/other-plans-reports/plain-language-page'
    }
  },
  plan: {
    en: {
      text: 'Plan'
    },
    es: {
      text: 'Planifique'
    }
  },
  policyAndRegulations: {
    en: {
      text: 'Policy and Regulations',
      url: '/about-sba/sba-performance/policy-regulations'
    },
    es: {
      text: 'Política y normativa',
      url: '/about-sba/sba-performance/policy-regulations'
    }
  },
  previous: {
    en: {
      text: 'Previous'
    },
    es: {
      text: 'Anterior'
    }
  },
  privacyPolicy: {
    en: {
      text: 'Privacy Policy',
      url: '/about-sba/sba-performance/open-government/about-sbagov-website/privacy-policy'
    },
    es: {
      text: 'Política de privacidad',
      url: '/about-sba/sba-performance/open-government/about-sbagov-website/privacy-policy'
    }
  },
  qualifyForGovernmentContracts: {
    en: {
      text: 'Qualify for Government Contracts',
      url: '/size-standards'
    },
    es: {
      text: 'Gane contratos del gobierno',
      url: '/size-standards'
    }
  },
  readMore: {
    en: 'Read more',
    es: 'Para saber más'
  },
  register: {
    en: {
      text: 'Register',
      url: '/user/register'
    },
    es: {
      text: 'Regístrese',
      url: '/user/register'
    }
  },
  regulationsGov: {
    en: {
      text: 'Regulations.gov',
      url: 'https://www.regulations.gov'
    },
    es: {
      text: 'Regulations.gov',
      url: 'https://www.regulations.gov'
    }
  },
  reportFraudWasteAndAbuse: {
    en: {
      text: 'Report Fraud, Waste, and Abuse',
      url: '/category/navigation-structure/contracting/contracting-officials/report-fraud-waste-abuse'
    },
    es: {
      text: 'Informe de fraude, dispendio y abuso',
      url: '/category/navigation-structure/contracting/contracting-officials/report-fraud-waste-abuse'
    }
  },
  research: {
    en: {
      text: 'Research'
    },
    es: {
      text: 'Investigue'
    }
  },
  sbaEnEspanol: {
    en: {
      text: 'SBA en Español',
      url: '/bienvenido-a-la-en-espanol'
    },
    es: {
      text: 'SBA en español',
      url: '/bienvenido-a-la-en-espanol'
    }
  },
  sbaInformation: {
    en: {
      text: 'SBA Information'
    },
    es: {
      text: 'Información de la SBA'
    }
  },
  sbaLocations: {
    en: {
      text: 'SBA Locations',
      url: '/about-sba/sba-locations'
    },
    es: {
      text: 'Ubicaciones de la SBA',
      url: '/about-sba/sba-locations'
    }
  },
  sbaNearYou: {
    en: {
      text: 'SBA Near You',
      url: '/tools/local-assistance#locations-page'
    },
    es: {
      text: 'SBA cerca de usted',
      url: '/tools/local-assistance#locations-page'
    }
  },
  sbaPerformance: {
    en: {
      text: 'SBA Performance',
      url: '/about-sba/sba-performance'
    },
    es: {
      text: 'Desempeño de la SBA',
      url: '/about-sba/sba-performance'
    }
  },
  sbaTeam: {
    en: {
      text: 'SBA Team',
      url: '/about-sba/sba-team'
    },
    es: {
      text: 'Equipo de la SBA',
      url: '/about-sba/sba-team'
    }
  },
  sbaVideos: {
    en: {
      text: 'SBA Videos',
      url: '/videos'
    },
    es: {
      text: 'Videos de la SBA',
      url: '/videos'
    }
  },
  select: {
    en: {
      text: 'Select...'
    },
    es: {
      text: 'Selecciona...'
    }
  },
  siteMap: {
    en: {
      text: 'Site Map',
      url: '/sitemap'
    },
    es: {
      text: 'Mapa del sitio',
      url: '/sitemap'
    }
  },
  smallBusinessEvents: {
    en: {
      text: 'Small Business Events',
      url: '/tools/events#events-page'
    },
    es: {
      text: 'Eventos para pequeñas empresas',
      url: '/tools/events#events-page'
    }
  },
  socialMedia: {
    en: {
      text: 'Social Media',
      url: '/about-sba/open-government/about-sbagov-website/social-media'
    },
    es: {
      text: 'Redes sociales',
      url: '/about-sba/open-government/about-sbagov-website/social-media'
    }
  },
  strategicPlanning: {
    en: {
      text: 'Strategic Planning',
      url: '/about-sba/sba-performance/strategic-planning'
    },
    es: {
      text: 'Planificación estratégica',
      url: '/about-sba/sba-performance/strategic-planning'
    }
  },
  structure: {
    en: {
      text: 'Structure'
    },
    es: {
      text: 'Estructura'
    }
  },
  taxIDs: {
    en: {
      text: 'Tax IDs'
    },
    es: {
      text: 'Identificaciones fiscales'
    }
  },
  toolsAndFeatures: {
    en: {
      text: 'Tools and Features'
    },
    es: {
      text: 'Herramientas y características'
    }
  },
  translate: {
    en: {
      text: 'Translate'
    },
    es: {
      text: 'Traducir'
    }
  },
  usaGov: {
    en: {
      text: 'USA.gov',
      url: 'https://www.usa.gov'
    },
    es: {
      text: 'USA.gov',
      url: 'https://www.usa.gov'
    }
  },
  whatWeDo: {
    en: {
      text: 'What We Do',
      url: '/about-sba/what-we-do'
    },
    es: {
      text: 'Nuestro trabajo',
      url: '/about-sba/what-we-do'
    }
  },
  whiteHouseGov: {
    en: {
      text: 'WhiteHouse.gov',
      url: 'https://www.whitehouse.gov'
    },
    es: {
      text: 'WhiteHouse.gov',
      url: 'https://www.whitehouse.gov'
    }
  },

  // The context where the translations are found are the keys:
  tenSteps1: {
    en: {
      title: 'Conduct market research',
      text:
        'Market research will tell you if there’s an opportunity to turn your idea into a successful business. It’s a way to gather information about potential customers and businesses already operating in your area. Use that information to find a competitive advantage for your business.',
      link: '/business-guide/plan/market-research-competitive-analysis'
    },
    es: {
      title: 'Haga una investigación de mercado',
      text:
        'Mediante la investigación de mercado podrá saber si hay oportunidad de convertir su idea en un negocio exitoso. Es una forma de reunir información sobre posibles clientes y empresas que ya funcionan en su zona. Utilice esa información para encontrarle una ventaja competitiva de su empresa.',
      link: '/guia-de-negocios/planifique-su-empresa/investigacion-de-mercado-y-analisis-competitivo'
    }
  },
  tenSteps2: {
    en: {
      title: 'Write your business plan',
      text:
        'Your business plan is the foundation of your business. It’s a roadmap for how to structure, run, and grow your new business. You’ll use it to convince people that working with you — or investing in your company — is a smart choice.',
      link: '/business-guide/plan-your-business/write-your-business-plan'
    },
    es: {
      title: 'Redacte su plan de negocios',
      text:
        'Su plan de negocios es la fundación de su empresa. Es un mapa de ruta de cómo estructurar, operar y hacer crecer su nueva empresa. Le servirá para convencer a la gente de que trabajar con usted (o invertir en su compañía) es una opción inteligente.',
      link: '/guia-de-negocios/planifique-su-empresa/redacte-su-plan-de-negocios'
    }
  },
  tenSteps3: {
    en: {
      title: 'Fund your business',
      text:
        'Your business plan will help you figure out how much money you’ll need to start your business. If you don’t have that amount on hand, you’ll need to either raise or borrow the capital. Fortunately, there are more ways than ever to find the capital you need.',
      link: '/business-guide/plan-your-business/fund-your-business'
    },
    es: {
      title: 'Financie su empresa',
      text:
        'Su plan de negocios le ayudará a calcular cuánto dinero necesitará para iniciar su empresa. Si no cuenta ahora con esa cantidad, tendrá que recaudar fondos o pedir prestado el capital. Afortunadamente, hoy hay más formas que nunca de encontrar el capital que necesita.',
      link: '/guia-de-negocios/planifique-su-empresa/financie-su-empresa'
    }
  },
  tenSteps4: {
    en: {
      title: 'Pick your business location',
      text:
        'Your business location is one of the most important decisions you’ll make. Whether you’re setting up a brick-and-mortar business or launching an online store, the choices you make could affect your taxes, legal requirements, and revenue.',
      link: '/business-guide/launch-your-business/pick-your-business-location'
    },
    es: {
      title: 'Escoja la ubicación de su empresa',
      text:
        'Una de las decisiones más importantes que tomará es la ubicación de su empresa. Sea que vaya a establecer una empresa con presencia física o a lanzar una tienda en línea, las decisiones que tome podrían influir en sus impuestos, sus obligaciones legales y sus ganancias.',
      link: '/guia-de-negocios/lance-su-empresa/escoja-la-ubicacion-de-su-empresa'
    }
  },
  tenSteps5: {
    en: {
      title: 'Choose a business structure',
      text:
        'The legal structure you choose for your business will impact your business registration requirements, how much you pay in taxes, and your personal liability.',
      link: '/business-guide/launch-your-business/choose-business-structure'
    },
    es: {
      title: 'Escoja una estructura empresarial',
      text:
        'La estructura legal que escoja para su negocio influirá en los requisitos de registro de su empresa, la cantidad de impuestos que pague y su responsabilidad personal.',
      link: '/guia-de-negocios/lance-su-empresa/escoja-una-estructura-empresarial'
    }
  },
  tenSteps6: {
    en: {
      title: 'Choose your business name',
      text:
        'It’s not easy to pick the perfect name. You’ll want one that reflects your brand and captures your spirit. You’ll also want to make sure your business name isn’t already being used by someone else.',
      link: '/business-guide/launch-your-business/choose-your-business-name'
    },
    es: {
      title: 'Elija el nombre de su empresa',
      text:
        'No es fácil escoger el nombre perfecto. Desearía uno que refleje su marca y que capte su espíritu. También querrá verificar que el nombre por el que se decidió no lo esté usando ya alguien.',
      link: '/guia-de-negocios/lance-su-empresa/elija-un-nombre-para-su-empresa'
    }
  },
  tenSteps7: {
    en: {
      title: 'Register your business',
      text:
        'Once you’ve picked the perfect business name, it’s time to make it legal and protect your brand. If you’re doing business under a name different than your own, you’ll need to register with the federal government, and maybe your state government, too.',
      link: '/business-guide/launch-your-business/register-your-business'
    },
    es: {
      title: 'Registre su empresa',
      text:
        'Una vez que haya escogido el nombre ideal para la empresa, será momento de legalizarlo y proteger su marca. Si está haciendo negocios con un nombre diferente de su nombre personal, deberá registrarse ante el gobierno federal y quizá también el gobierno de su estado.',
      link: '/guia-de-negocios/lance-su-empresa/registre-su-empresa'
    }
  },
  tenSteps8: {
    en: {
      title: 'Get federal and state tax IDs',
      text:
        'You’ll use your employer identification number (EIN) for important steps to start and grow your business, like opening a bank account and paying taxes. It’s like a social security number for your business. Some — but not all — states require you to get a tax ID as well.',
      link: '/business-guide/launch-your-business/get-federal-and-state-tax-id-numbers'
    },
    es: {
      title: 'Obtenga su identificación fiscal federal y estatal',
      text:
        'Utilizará su número de identificación de empleador (EIN) en acciones importantes para iniciar y hacer crecer su empresa, como abrir una cuenta bancaria y pagar impuestos. Es como un número del seguro social para su empresa. Algunos estados también piden obtener una identificación fiscal.',
      link:
        '/guia-de-negocios/lance-su-empresa/obtenga-su-numero-de-identificacion-del-contribuyente-federal-y-estatal'
    }
  },
  tenSteps9: {
    en: {
      title: 'Apply for licenses and permits',
      text:
        'Keep your business running smoothly by staying legally compliant. The licenses and permits you need for your business will vary by industry, state, location, and other factors.',
      link: '/business-guide/launch-your-business/apply-licenses-and-permits'
    },
    es: {
      title: 'Solicite licencias y permisos',
      text:
        'Cumpla siempre con las normas para que su negocio se mantenga funcionando sin tropiezos. Las licencias y permisos que necesite su empresa varían con el sector en que opere, el estado, la ubicación y otros factores.',
      link: '/guia-de-negocios/lance-su-empresa/solicite-licencias-y-permisos'
    }
  },
  tenSteps10: {
    en: {
      title: 'Open a business bank account',
      text:
        'A small business checking account can help you handle legal, tax, and day-to-day issues. The good news is it’s easy to set one up if you have the right registrations and paperwork ready.',
      link: '/business-guide/launch-your-business/open-business-bank-account'
    },
    es: {
      title: 'Abra una cuenta bancaria comercial',
      text:
        'Una cuenta de cheques para pequeñas empresas puede ayudarle a manejar cuestiones legales, fiscales y del día a día. La buena noticia es que es fácil abrir una si tiene listos los registros y documentos correctos.',
      link: '/guia-de-negocios/lance-su-empresa/abra-una-cuenta-bancaria-comercial'
    }
  },
  tenStepsHero: {
    en: {
      title: '10 steps to start your business',
      text:
        'Starting a business involves planning, making key financial decisions, and completing a series of legal activities. Scroll down to learn about each step.',
      link: '/business-guide'
    },
    es: {
      title: '10 pasos para iniciar su empresa',
      text:
        'Emprender un negocio requiere planificar, tomar decisiones financieras cruciales y realizar una serie de actividades legales. Desplácese hacia abajo para conocer cada paso.',
      link: '/guia-de-negocios'
    }
  },
  tenStepsNavigationCta: {
    en: {
      headline: 'Not sure where to start? Start your business in 10 steps.',
      link: '/business-guide/10-steps-start-your-business/',
      linkText: 'See the guide'
    },
    es: {
      headline: '¿No sabe por dónde empezar? Siga estos 10 pasos.',
      link: '/guia-de-negocios/10-pasos-para-iniciar-su-empresa',
      linkText: 'Vea la guía'
    }
  },
  tenStepsOpenShop: {
    en: {
      title: 'Open Shop',
      text:
        "Congratulations ! It's time to cut the big ribbon. Your business is officially open. Now, focus on managing and growing your business."
    },
    es: {
      title: 'Empresa abierta',
      text:
        '¡Felicidades! Llegó la hora de cortar la cinta. Su negocio está oficialmente establecido. Ahora, concéntrese en administrarlo y hacerlo crecer.'
    }
  },
  tenStepsPanelCta: {
    en: {
      headline: 'Start your business in 10 steps',
      link: '/business-guide/10-steps-start-your-business/',
      linkText: 'See the guide'
    },
    es: {
      headline: 'Abra su empresa siguiendo estos 10 pasos',
      link: '/guia-de-negocios/10-pasos-para-iniciar-su-empresa',
      linkText: 'Vea la guía'
    }
  }
}

export { TRANSLATIONS }
