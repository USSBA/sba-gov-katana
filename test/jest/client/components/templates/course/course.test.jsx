import React from 'react'
var sinon = require('sinon')
import { shallow } from 'enzyme'
import * as helper from 'client/fetch-content-helper'
import Course from 'templates/course/course'
let exampleCourse = {
  body:
    '<p>This course explains the importance of business planning, defines and describes the components of a business plan, and provides access to sample plans and resources that can help you develop a very good business plan.<br />\r\nDuration: 00:30:00</p>\r\n',
  courseCategory: ['Plan your business'],
  courseMaterials: [
    {
      url: '/sites/default/files/2018-01/CHECKLIST_-_How_to_Write_a_Business_Plan.pdf',
      description: 'Checklist - How to Write a Business Plan'
    },
    {
      url: '/sites/default/files/2018-01/SBA_Keyboard_Shortcuts_6_13.pdf',
      description: 'SBA Keyboard Shortcuts'
    },
    {
      url: '/sites/default/files/2018-01/BalanceSheet.xlsx',
      description: 'Balance Sheet Template'
    },
    {
      url: '/sites/default/files/2018-01/Income_Statement_for_BPlan_Course.xlsx',
      description: 'Income Statement Template'
    },
    {
      url: '/sites/default/files/2018-01/Cash_Flow_Statement.xlsx',
      description: 'Cash Flow Statement Template'
    }
  ],
  courseType: 'Flash',
  courseUrl: {
    url: '/media/training/howtowriteabusinessplan_02022017/story.html',
    title: ''
  },
  image: {
    url: '/sites/default/files/2018-01/2-plan_500.jpg',
    alt: 'How to Write a Business Plan',
    width: 500,
    height: 340
  },
  summary:
    'Learn the importance of business planning, the components of a business plan, and see sample plans and resources.',
  transcript: 'Some Transcript',
  type: 'course',
  title: 'How to Write a Business Plan',
  id: 5880,
  updated: 1519147473,
  created: 1516892830,
  url: '/course/how-write-business-plan',
  isFound: true
}

describe('Course', () => {
  let fetchSiteContentStub
  beforeAll(done => {
    fetchSiteContentStub = sinon.stub(helper, 'fetchSiteContent')
    done()
  })

  afterAll(done => {
    fetchSiteContentStub.restore()
    done()
  })

  beforeEach(done => {
    history.pushState({}, null, '?lang=en')
    fetchSiteContentStub.reset()
    done()
  })

  test('rendering of the CourseContent', () => {
    fetchSiteContentStub.returns(Promise.resolve(exampleCourse))
    const component = shallow(<Course location={{ pathname: '/test/course' }} />)
    expect(component).toMatchSnapshot()
  })
})
