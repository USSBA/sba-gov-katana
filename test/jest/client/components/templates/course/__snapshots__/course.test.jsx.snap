// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course rendering of the CourseContent 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <CourseView
      title="My Course Title"
    />
  </div>,
  "nodes": Array [
    <div>
      <CourseView
        title="My Course Title"
      />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Course
        location={
          Object {
            "pathname": "/test/course",
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Course {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "location": Object {
            "pathname": "/test/course",
          },
        },
        "refs": Object {},
        "state": Object {
          "title": "My Course Title",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <CourseView
            title="My Course Title"
          />
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <CourseView
            title="My Course Title"
          />
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Course
    location={
      Object {
        "pathname": "/test/course",
      }
    }
  />,
}
`;
